src/library/scala/concurrent/ExecutionContext.scala

```diff
@@ -166,14 +170,60 @@ object ExecutionContext {
     override final def reportFailure(t: Throwable): Unit = defaultReporter(t)
   }
 
+  /**
+   * This `ExecutionContext` uses the same thread pool as [[ExecutionContext.global]]. It attempts to batch nested
+   * task and execute them on the same thread as the enclosing task. This is ideally suited to execute short-lived tasks
+   * as it reduces the overhead of context switching.
+   *
+   * WARNING: long-running and/or blocking tasks should be demarcated within [[scala.concurrent.blocking]]-blocks
+   *          to ensure that any pending tasks in the current batch can be executed by another thread on `global`.
+   *
+   * === How to use ===
+   *
+   * This `ExecutionContext` is `private[scala]` to maintain binary compatibility. It was added in 2.13.3, code that
+   * uses it will fail to run with a 2.13.0-2 Scala library.
+   *
+   * Libraries should not reference this field because users of the library might be using an earlier Scala version.
+   *
+   * Application authors can safely use the field because the Scala version at run time is the same as at compile time.
+   * Options to bypass the access restriction include:
+   *
+   *   1. Using a structural type (see below). This uses reflection at run time.
+   *   1. Writing a Scala `object` in the `scala` package (see below).
+   *   1. Writing a Java source file. This works because `private[scala]` is emitted as `public` in Java bytecode.
+   *
+   * {{{
+   * // Option 1
+   * implicit lazy val ec: ExecutionContext =
```

Comment:
@lrytz Perhaps we should mention that the access should be done with a fallback to `global` in case it isn't available?

Reply:
Ah that's a good point! We can add a third option:

```
implicit val ec: scala.concurrent.ExecutionContext = try {
  scala.concurrent.ExecutionContext.getClass
    .getDeclaredMethod("opportunistic")
    .invoke(scala.concurrent.ExecutionContext)
    .asInstanceOf[scala.concurrent.ExecutionContext]
} catch {
  case _: NoSuchMethodException =>
    scala.concurrent.ExecutionContext.global
}
```

This is safe to use also in a library and returns the batching EC on all 2.13 versions.

Reply:
Yep!